Switch `develop` to default branch and add `Merging Processes` to contributing.md (#552)
trufflesuite/trufflesuite.com: CONTRIBUTING.md

* [Search](#search)
* [Translating](#translating)

## Branching Processes

Pull requests should be made against the `develop` branch.

This repo uses short-lived, prefixed branches. For example, a new blog post would live a branch named something like: `content/cool-truffle-news`. This helps multiple teams touch the site if necessary, while mitigating potential collisions. These are the currently used prefixes:

`content`: edits applying to site content.
`feature`: new features for the website, including new data structures (e.g. blog posts or events).

> `fix`: fixes an error in existing code.

Reviewer: Nice process! and description!

## Merging Processes

1. As mentioned above, your contribution will be a branch based off of `develop`
1. Once it's ready for review, please create a new [Pull Request](https://github.com/trufflesuite/trufflesuite.com/compare) to be merged into (aka base branch) `develop`

> 1. PRs from Truffle employee require at least one review approval, PRs from external contributors are recommended (though not required) to have >1 review

Reviewer: "Member" is probably a better noun than "Truffle employee" (more consistent-with-GH-terms), if it's easy to make sound alright

Reviewer: ah also: two separate sentences are joined by semicolons:
```suggestion
1. PRs from Truffle employee require at least one review approval; PRs from external contributors are recommended (though not required) to have >1 review
```

comma alternative:
```suggestion
1. PRs from Truffle employee require at least one review approval, and PRs from external contributors are recommended (though not required) to have >1 review
```

> 1. PRs that passes all automatic check **and** meets the approval criteria above may be merged by anyone (though give the author time to merge it themselves as they deserve the satisfaction of pushing the button)

Reviewer: emoji maybe?

Reviewer: also grammar 
```suggestion
1. PRs that pass all automatic checks **and** meet the approval criteria above may be merged by anyone (though give the author time to merge it themselves, as they deserve the satisfaction of pushing the button)
```

>   - After a PR has been merge, **please delete the original branch**

Reviewer: verb tense
```suggestion
  - After a PR has been merged, **please delete the original branch**
```

> 1. When preparing to publish to master, anyone may merge `develop` into `master` **after** successfully testing in the staging environment.

Reviewer: It may or may not be worth mentioning, also, that `master` **must** be safely mergeable into `develop` at any time, with the general rule that it active branches **should** pull from active upstream every 24 hours.

if you want to make that sound like not a mouthful

Author: I think stating this is a little overkill and not clear to understand by all
